specs/sharding/beacon-chain.md

```diff
@@ -705,20 +731,46 @@ def process_shard_header(state: BeaconState, signed_header: SignedShardBlobHeade
         == bls.Pairing(body_summary.commitment.point, G2_SETUP[-body_summary.commitment.length])
     )
 
-    # Charge builder, with hard balance requirement
-    fee = Gwei(123)  # TODO EIP 1559 like fee? Burn some of it?
-    charge_builder(state, header.builder_index, fee)
-    # TODO: proposer is charged for confirmed headers (see charge_confirmed_shard_fees).
-    #  Need to align incentive, so proposer does not gain from including unconfirmed headers
-    increase_balance(state, header.proposer_index, fee)
+    # Charge EIP 1559 fee, builder pays for opportunity, and is responsible for later availability,
+    # or fail to publish at their own expense.
+    samples = blob_summary.commitment.length
+    # TODO: overflows, need bigger int type
+    max_fee = blob_summary.max_fee_per_sample * samples
+
+    # Builder must have sufficient balance, even if max_fee is not completely utilized
+    assert state.blob_builder_balances[header.builder_index] > max_fee
```

Comment:
Shouldn't it be `>=` instead of `>` here?

Reply:
Good catch